p2p/net/upgrader/multistream.go

```diff
@@ -24,21 +33,21 @@ type Transport struct {
 	OrderPreference []string
 }
 
-func NewBlankTransport() *Transport {
+func NewMsTransport() MsTransport {
 	return &Transport{
 		mux:              mss.NewMultistreamMuxer(),
 		tpts:             make(map[string]network.Multiplexer),
 		NegotiateTimeout: DefaultNegotiateTimeout,
 	}
 }
 
-func (t *Transport) AddTransport(path string, tpt network.Multiplexer) {
+func (t *Transport) AddMuxer(path string, tpt network.Multiplexer) {
 	t.mux.AddHandler(path, nil)
 	t.tpts[path] = tpt
 	t.OrderPreference = append(t.OrderPreference, path)
 }
 
-func (t *Transport) NewConn(nc net.Conn, isServer bool, scope network.PeerScope) (network.MuxedConn, error) {
+func (t *Transport) NegotiateMuxer(nc net.Conn, isServer bool) (*Multiplexer, error) {
```

Comment:
This is the only function you need – and it's just a thin wrapper around `go-multistream`. I would say:

1. we move this to the upgrader (No one else really cares about this package (I think)).
2. Use a ctx to set the negotiate timeout
3. Make it private
4. Delete this whole package. less code!